include/bluetooth/audio/audio.h

```diff
@@ -1514,6 +1514,45 @@ int bt_audio_unicast_group_create(struct bt_audio_stream *streams,
 				  uint8_t num_stream,
 				  struct bt_audio_unicast_group **unicast_group);
 
+/** @brief Add streams to a unicast group as a unicast client
+ *
+ *  This function can be used to add additional streams to a
+ *  bt_audio_unicast_group.
+ *
+ *  This can be called at any time before any of the streams in the
+ *  group has been started (see bt_audio_stream_ops.started()).
+ *  This can also be called after the streams have been stopped
+ *  (see bt_audio_stream_ops.stopped()).
+ *
+ *  @param unicast_group  Pointer to the unicast group
+ *  @param streams        Array of stream objects being added to the group.
+ *  @param num_stream     Number of streams in @p streams.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_audio_unicast_group_add_streams(struct bt_audio_unicast_group *unicast_group,
+				       struct bt_audio_stream *streams,
+				       uint8_t num_stream);
+
+/** @brief Remove streams from a unicast group as a unicast client
+ *
+ *  This function can be used to remove streams from a bt_audio_unicast_group.
+ *
+ *  This can be called at any time before any of the streams in the
+ *  group has been started (see bt_audio_stream_ops.started()).
+ *  This can also be called after the streams have been stopped
+ *  (see bt_audio_stream_ops.stopped()).
+ *
+ *  @param unicast_group  Pointer to the unicast group
+ *  @param streams        Array of stream objects removed from the group.
+ *  @param num_stream     Number of streams in @p streams.
+ *
+ *  @return 0 in case of success or negative value in case of error.
+ */
+int bt_audio_unicast_group_remove_streams(struct bt_audio_unicast_group *unicast_group,
+					  struct bt_audio_stream *streams,
+					  uint8_t num_stream);
```

Comment:
`uint8_t` seems a bit arbitrary. Why not eg `size_t` as that’s the return value of `ARRAY_SIZE()`?

Reply:
Used `uint8_t` as the maximum number of streams is 0x1F (as per the Core spec). 

I am fine with using size_t, but I will also add the maximum number of streams to the documentation (regardless of uint8_t/size_t) as that is missing.

Reply:
Actually, it is already implicitly described by CONFIG_BT_AUDIO_UNICAST_CLIENT_GROUP_STREAM_COUNT, so won't add additional documentation (unless requested).

Reply:
I think I’d still prefer size_t since in the API context this relates to the previous array parameter and not over the air representation. And I suppose the functions anyway check for the max value and return error if too large.

Reply:
Btw, do we have some convention of using `array[]` instead of `*array` to make it clear the parameter is an array and not a pointer to a single element?

Reply:
> Btw, don’t we have some convention of using `array[]` instead of `*array` to make it clear the parameter is an array and not a pointer to a single element?

We do not. I actually don't see any BT APIs use `array[]`.

Reply:
Good point. I took a look at e.g. `bt_le_adv_start()` (one of the "old" APIs we have), and it uses pointer syntax. Still, I don't see any reasons not to make the distinction clearer by using `[]`, but that'd then need to be done consistently throughout the subsystem (i.e. independent of this PR).

Reply:
> The value is used for a HCI request that takes an uint8_t however, so we'll need to add a cast later on if we pursue the size_t route.

Why would you need an (explicit) cast? I don't think the compiler will complain of an implicit case from size_t to uint8_t, will it?

Reply:
Furthermore, variable types of members of packed structs (such as HCI command parameters) should in many ways be thought of as belonging to a separate domain because of different alignment and byte order (for multi-byte types) assumptions, meaning one always needs to pay some attention when encoding to/from them.

Reply:
> Do you mind if I create an issue to fix this in a separate PR? This will also affect the ISO API which also use uint8_t for array counts.

Can't the ISO API be updated in a separate PR?

Reply:
Btw, can you point me to the specific APIs in the ISO API using uint8_t for array sizes?

Reply:
> Btw, can you point me to the specific APIs in the ISO API using uint8_t for array sizes?

```c
struct bt_iso_cig_param {
	/** @brief Array of pointers to CIS channels */
	struct bt_iso_chan **cis_channels;

	/** @brief Number channels in @p cis_channels
	 *
	 *  Maximum number of channels in a single group is
	 *  BT_ISO_MAX_GROUP_ISO_COUNT
	 */
	uint8_t num_cis;
```

Reply:
> > Do you mind if I create an issue to fix this in a separate PR? This will also affect the ISO API which also use uint8_t for array counts.
> 
> Can't the ISO API be updated in a separate PR?

Sure, we can also just do it that way :)

Reply:
@Thalley thanks.

What are your thoughts on stuff like `struct bt_iso_chan **cis_channels` and the array vs pointer syntax. Wouldn't the type be clearer as `struct bt_iso_chan *cis_channels[]`?

Reply:
Btw, regarding size_t vs uint8_t - it's not completely the same with ISO. There you have the additional consideration of struct packing. However, with the current layout of `bt_iso_cig_parem` there will not be any memory penalty with the changed type.

Reply:
@jhedberg Added a new commit that changes the uses of uint8_t to size_t when using describing an array length.

Reply:
> @Thalley thanks.
> 
> What are your thoughts on stuff like `struct bt_iso_chan **cis_channels` and the array vs pointer syntax. Wouldn't the type be clearer as `struct bt_iso_chan *cis_channels[]`?

I may not be representative, but I find the latter clearer. (And ditto for [] for an array.)